### PR TITLE
Increase options.go coverage to 100%

### PR DESCRIPTION
**Pull request by krader1961**
Related #1062

### CODE REVIEW

**Comment by krader1961** [comment]
The "Test on Linux" environment failed with this error:

```
--- FAIL: TestLocationAddon_Workspace (6.46s)
    listing_nonwindows_test.go:73: Wanted buffer not shown
    listing_nonwindows_test.go:73: Want: Width = 50, Dot = (0, 11)
        ┌──────────────────────────────────────────────────┐
        │~/ws1/bin> $                                      │
        └──────────────────────────────────────────────────┘
    listing_nonwindows_test.go:73: Last buffer: Width = 50, Dot = (1, 11)
        ┌──────────────────────────────────────────────────┐
        │~/ws1/tmp> $                                      │
        │�[1;37;45m LOCATION �[m $                                      │
        │�[7m 10 ws/bin                                        �[m│
        │ 10 /usr/bin$                                     │
        └──────────────────────────────────────────────────┘
FAIL
FAIL	github.com/elves/elvish/pkg/edit	86.464s
```

That's a failure I'm seeing very infrequently on my local systems. See https://github.com/elves/elvish/issues/997 where a very similar failure was discussed. It looks like this is another test that needs to be given more time before reporting a failure.

**Comment by xiaq** [comment]
Hmm, this particular failing test depends on some IO (changing directory), so it's no wonder that among all the tests this one fails. In fact, looking at #997 the one that failed for you also requires some IO (getting current directory).

In the long term these tests can be made more deterministic by using an IO abstraction layer in the spirit of [afero](https://godoc.org/github.com/spf13/afero) (afero doesn't abstract pwd, though). In the meanwhile I'll just bump the test time scale even further to 20.

File: pkg/eval/options_test.go [hunk 1]
```diff
@@ -1,28 +1,45 @@
...
+		t.Errorf("Scan should have reported invalid arg")
+	}
+
 	for _, test := range scanOptionsTests {
 		opts := test.preScan
-		scanOptions(test.rawOpts, &opts)
+		err := scanOptions(test.rawOpts, &opts)
+		if err != nil && test.err == nil {
+			t.Errorf("Scan %v => %v, unexpected error %v", test.rawOpts, opts, err)
+		}
```
Comment (xiaq) [inline pkg/eval/options_test.go:42]: * This isn't checking against the actual value of `test.err`. Either check that (you can check the error message), or just specify a boolean in the test struct.

* This doesn't cover the case where `err == nil && test.err != nil`.

Comment (krader1961) [reply pkg/eval/options_test.go:42]: Yeah, I was taking an unwarranted shortcut based on the knowledge there is currently only one error condition. Fixed so the test is now future proof.

**Pull request merged**